### PR TITLE
Fix useAllPaginated query C-2980

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Fixes a few issues with useAllPaginated

-Data would load in repeatedly + infinitely due to repeated `loadMore` calls
-Data would not load in at all after it was reset (e.g. due to a baseArgs change)

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--fix-paginated-query-r

File: packages/common/src/audius-query/hooks/usePaginatedQuery.ts [hunk 1]
```diff
@@ -64,28 +65,52 @@ export const useAllPaginatedQuery = <
...
   useCustomCompareEffect(
     () => {
       setAllData([])
+      setLoadingMore(false)
     },
     [baseArgs],
-    isEqual
+    (prevDeps, nextDeps) => {
+      return isEqual(prevDeps, nextDeps)
+    }
```
Comment (amendelsohn) [inline packages/common/src/audius-query/hooks/usePaginatedQuery.ts:77]: does `isEqual` work here (and below) without the wrapper?

File: packages/common/src/audius-query/hooks/usePaginatedQuery.ts [hunk 2]
```diff
@@ -64,28 +65,52 @@ export const useAllPaginatedQuery = <
...
+    result.status !== Status.ERROR &&
+    ((result.status === Status.IDLE && allData.length === 0) ||
+      (!result.data && result.status !== Status.SUCCESS) ||
+      result.data?.length === pageSize)
+
+  const loadMore = useCallback(() => {
+    if (loadingMore) {
+      return
+    }
+    setLoadingMore(true)
```
Comment (amendelsohn) [inline packages/common/src/audius-query/hooks/usePaginatedQuery.ts:104]: ah this is a good adaptation. Nice!

**Review by amendelsohn — APPROVED** [review]
Thanks so much for fixing this!!

File: packages/common/src/audius-query/hooks/usePaginatedQuery.ts [hunk 3]
```diff
@@ -64,28 +65,52 @@ export const useAllPaginatedQuery = <
...
+      if (result.status !== Status.SUCCESS) return
+      setAllData((allData) => [...allData, ...result.data])
+    },
+    [result.status, args],
+    (prevDeps, nextDeps) => {
+      return isEqual(prevDeps, nextDeps)
+    }
   )
 
+  const hasMore =
```
Comment (raymondjacobson) [inline packages/common/src/audius-query/hooks/usePaginatedQuery.ts:90]: nit for readability

```
const notError = result.status !== Status.ERROR
const idleAndNoData = result.status === Status.IDLE && allData.length === 0
const hasNotFetched = !result.data && result.status !== Status.SUCCESS
const fetchedFullPreviousPage = result.data?.length === pageSize

const hasMore = notError && (idleAndNoData || hasNotFetched || fetchedFullPreviousPage)
```

Not quite sure if I understand the `idleAndNoData` case though

Comment (nicoback2) [reply packages/common/src/audius-query/hooks/usePaginatedQuery.ts:90]: ty great suggestion! idle + no data is just the initial state before anything loads

**Pull request merged**